File: server_app/src/api/models/report/report.go
package report

import (
	"time"

	"gorm.io/gorm"
)

type Report struct {
	gorm.Model
	ID        string    `gorm:"type:uuid;default:gen_random_uuid();unique"`
	ServiceID string    `gorm:"NOT NULL;type:uuid"`
	UserID    string    `gorm:"NOT NULL;type:uuid"`
	Reason    string    `gorm:"NOT NULL;size:255"`
	CreatedAt time.Time `gorm:"NOT NULL"`
	Status    bool      `gorm:"NOT NULL;DEFAULT:false"`
}

File: front_app/lib/features/admin_app/models/report.dart
class Report {
  String id;
  String userId;
  String serviceId;
  String reason;
  DateTime createdAt;
  bool status;

  Report({
    required this.id,
    required this.userId,
    required this.serviceId,
    required this.reason,
    required this.createdAt,
    required this.status,
  });

  factory Report.fromJson(Map<String, dynamic> json) => Report(
        id: json['id'],
        userId: json['userId'],
        serviceId: json['serviceId'],
        reason: json['reason'],
        createdAt: DateTime.parse(json['createdAt']),
        status: json['status'],
      );

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'serviceId': serviceId,
      'reason': reason,
      'createdAt': createdAt.toIso8601String(),
      'status': status,
    };
  }
}
